src/components/ai-input.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useRef, useEffect } from 'react';
+import Button from './button';
+import { AssistantIcon } from './icons/assistant';
+import { MenuIcon } from './icons/menu';
+import { ReturnIcon } from './icons/return';
+
+interface AIInputProps {
+	disabled: boolean;
+	input: string;
+	setInput: ( input: string ) => void;
+	handleSend: () => void;
+	handleKeyDown: ( e: React.KeyboardEvent< HTMLTextAreaElement > ) => void;
+	clearInput: () => void;
+	isAssistantThinking: boolean;
+}
+
+const MAX_ROWS = 10;
+
+export const AIInput: React.FC< AIInputProps > = ( {
+	disabled,
+	input,
+	setInput,
+	handleSend,
+	handleKeyDown,
+	clearInput,
+	isAssistantThinking,
```

Comment:
I've observed a lint error about `isAssistantThinking` not being used, but I think we'll use it in the short term.

Reply:
Correct -- I left it in as I assumed we'd be using it for either the star animation, or updating the placeholder while generating,.